diff --git a/devel.module b/devel.module
index 7c4175c..4b1651d 100644
--- a/devel.module
+++ b/devel.module
@@ -468,6 +468,7 @@ function has_krumo() {
   @include_once DRUPAL_ROOT . '/' . drupal_get_path('module', 'devel') . '/krumo/class.krumo.php';
   if (function_exists('krumo') && !drupal_is_cli()) {
     drupal_add_js(drupal_get_path('module', 'devel') . '/devel_krumo_path.js');
+    drupal_add_js(drupal_get_path('module', 'devel') . '/krumo/krumo.js');
     return TRUE;
   }
   return FALSE;
diff --git a/krumo/krumo.js b/krumo/krumo.js
index 240af8c..5ae7044 100755
--- a/krumo/krumo.js
+++ b/krumo/krumo.js
@@ -22,7 +22,7 @@ function krumo() {
 * @return void
 */
 krumo.reclass = function(el, className) {
-	if (el.className.indexOf(className) < 0) {
+	if (typeof el.className !== 'undefined' && el.className.indexOf(className) < 0) {
 		el.className += (' ' + className);
 		}
 	}
@@ -37,7 +37,7 @@ krumo.reclass = function(el, className) {
 * @return void
 */
 krumo.unclass = function(el, className) {
-	if (el.className.indexOf(className) > -1) {
+	if (typeof el.className !== 'undefined' && el.className.indexOf(className) > -1) {
 		el.className = el.className.replace(className, '');
 		}
 	}
